### PR TITLE
Patch Endpoint 

### PR DESCRIPTION
**Pull request by badekk**
Create new endpoint to patch user details

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?

- [ ] Unit test(s)
- [X] Integration test(s)
- [X] Tested manually

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have checked my code and corrected any misspellings
- [X] I have added frontend code and checked it for mobile & desktop

### CODE REVIEW

**Comment by mar-pan** [comment]
Task linked: [CU-862hy9urf PATCH endpoint](https://app.clickup.com/t/862hy9urf)

File: src/backend/AuthService/BusinessLogic/PatchUserDetails/PatchUserDetailsService.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        private readonly AppDbContext _appDbContext;
+        private readonly ILogger<GetAuth0UsersByCurrentUserService> _logger;
+
+        public PatchUserDetailsService(AppDbContext appDbContext, ILogger<GetAuth0UsersByCurrentUserService> logger)
+        {
+            _appDbContext = appDbContext;
+            _logger = logger;
+        }
+
+        public async Task<bool> PatchUserFirstNameAndLastName(PatchUserDetailsPayload userSettingsPayload, long id)
```
Comment (h4570) [inline src/backend/AuthService/BusinessLogic/PatchUserDetails/PatchUserDetailsService.cs:19]: `Task<bool>` na `Task` - poza tym gra. 🎆

File: src/backend/AuthService/BusinessLogic/PatchUserDetails/PatchUserDetailsAction.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            _logger = logger;
+            _patchUserDetailsService = updateUserDetailsService;
+            _patchUserDetailsPayloadDataValidationService = patchUserDetailsPayloadDataValidationService;
+        }
+
+        public async Task<ObjectResult> Execute(PatchUserDetailsPayload userSettingsPayload, long id)
+        {
+            var patchUserDetailsPayloadIsValid = _patchUserDetailsPayloadDataValidationService.Check(userSettingsPayload);
+
+            if (!patchUserDetailsPayloadIsValid)
```
Comment (h4570) [inline src/backend/AuthService/BusinessLogic/PatchUserDetails/PatchUserDetailsAction.cs:28]: Taka opcjonalna zmiana.
Jest drugie podejście do czegoś takiego, w którym możesz przenieść tego ifa wraz z zawartością do `ValidationService`. Wtedy tutaj się trochę odchudzi i `ValidationService.Check()` zamiast zwracać `bool`, będzie mógł być void.

**Review by h4570 — CHANGES_REQUESTED** [review]
Ładnie ci poszło

File: src/backend/AuthService/BusinessLogic/PatchUserDetails/PatchUserDetailsPayloadDataValidationService.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AuthService.BusinessLogic.PatchUserDetails
+{
+    public class PatchUserDetailsPayloadDataValidationService
+    {
+        public bool Check(PatchUserDetailsPayload userDetailsPayload)
+        {
+            var check = userDetailsPayload.FirstName != null;
```
Comment (h4570) [inline src/backend/AuthService/BusinessLogic/PatchUserDetails/PatchUserDetailsPayloadDataValidationService.cs:14]: A jeszcze mi wpadło do głowy. 
Zwaliduj tutaj, czy `FirstName` i `LastName` ma długość `<=` niż pole z bazy. Rzuć tu wyjątkiem jak będzie źle.

**Review by h4570 — APPROVED** [review]

**Pull request merged**